// @gadget-client/shopappchat/dist-esm/models/AnalyticsFIVE.js
import { buildModelManager } from "../builder.js";
const DefaultAnalyticsFIVESelection = {
  __typename: true,
  id: true,
  createdAt: true,
  distinctId: true,
  event: true,
  properties: true,
  sessionId: true,
  shopId: true,
  timestamp: true,
  updatedAt: true,
  userId: true
};
const modelApiIdentifier = "analyticsFIVE";
const pluralModelApiIdentifier = "analyticsFIVEs";
;
;
;
;
;
;
;
;
;
const operations = [
  {
    type: "findOne",
    operationName: modelApiIdentifier,
    modelApiIdentifier,
    findByVariableName: "id",
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "maybeFindOne",
    operationName: modelApiIdentifier,
    modelApiIdentifier,
    findByVariableName: "id",
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "findMany",
    operationName: pluralModelApiIdentifier,
    modelApiIdentifier,
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "findFirst",
    operationName: pluralModelApiIdentifier,
    modelApiIdentifier,
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "maybeFindFirst",
    operationName: pluralModelApiIdentifier,
    modelApiIdentifier,
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "findOne",
    operationName: pluralModelApiIdentifier,
    functionName: "findById",
    findByField: "id",
    findByVariableName: "id",
    modelApiIdentifier,
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "maybeFindOne",
    operationName: pluralModelApiIdentifier,
    functionName: "maybeFindById",
    findByField: "id",
    findByVariableName: "id",
    modelApiIdentifier,
    defaultSelection: DefaultAnalyticsFIVESelection,
    namespace: null
  },
  {
    type: "action",
    operationName: "createAnalyticsFIVE",
    operationReturnType: "CreateAnalyticsFIVE",
    functionName: "create",
    namespace: null,
    modelApiIdentifier,
    operatesWithRecordIdentity: false,
    modelSelectionField: modelApiIdentifier,
    isBulk: false,
    isDeleter: false,
    variables: {
      analyticsFIVE: { required: false, type: "CreateAnalyticsFIVEInput" }
    },
    hasAmbiguousIdentifier: false,
    paramOnlyVariables: [],
    hasReturnType: false,
    acceptsModelInput: true,
    hasCreateOrUpdateEffect: true,
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "action",
    operationName: "bulkCreateAnalyticsFIVEs",
    functionName: "bulkCreate",
    isBulk: true,
    isDeleter: false,
    hasReturnType: false,
    acceptsModelInput: true,
    operatesWithRecordIdentity: false,
    singleActionFunctionName: "create",
    modelApiIdentifier,
    modelSelectionField: pluralModelApiIdentifier,
    namespace: null,
    variables: {
      inputs: { required: true, type: "[BulkCreateAnalyticsFIVEsInput!]" }
    },
    paramOnlyVariables: [],
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "action",
    operationName: "updateAnalyticsFIVE",
    operationReturnType: "UpdateAnalyticsFIVE",
    functionName: "update",
    namespace: null,
    modelApiIdentifier,
    operatesWithRecordIdentity: true,
    modelSelectionField: modelApiIdentifier,
    isBulk: false,
    isDeleter: false,
    variables: {
      id: { required: true, type: "GadgetID" },
      analyticsFIVE: { required: false, type: "UpdateAnalyticsFIVEInput" }
    },
    hasAmbiguousIdentifier: false,
    paramOnlyVariables: [],
    hasReturnType: false,
    acceptsModelInput: true,
    hasCreateOrUpdateEffect: true,
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "action",
    operationName: "bulkUpdateAnalyticsFIVEs",
    functionName: "bulkUpdate",
    isBulk: true,
    isDeleter: false,
    hasReturnType: false,
    acceptsModelInput: true,
    operatesWithRecordIdentity: true,
    singleActionFunctionName: "update",
    modelApiIdentifier,
    modelSelectionField: pluralModelApiIdentifier,
    namespace: null,
    variables: {
      inputs: { required: true, type: "[BulkUpdateAnalyticsFIVEsInput!]" }
    },
    paramOnlyVariables: [],
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "action",
    operationName: "deleteAnalyticsFIVE",
    operationReturnType: "DeleteAnalyticsFIVE",
    functionName: "delete",
    namespace: null,
    modelApiIdentifier,
    operatesWithRecordIdentity: true,
    modelSelectionField: modelApiIdentifier,
    isBulk: false,
    isDeleter: true,
    variables: { id: { required: true, type: "GadgetID" } },
    hasAmbiguousIdentifier: false,
    paramOnlyVariables: [],
    hasReturnType: false,
    acceptsModelInput: false,
    hasCreateOrUpdateEffect: false,
    defaultSelection: null
  },
  {
    type: "action",
    operationName: "bulkDeleteAnalyticsFIVEs",
    functionName: "bulkDelete",
    isBulk: true,
    isDeleter: true,
    hasReturnType: false,
    acceptsModelInput: false,
    operatesWithRecordIdentity: true,
    singleActionFunctionName: "delete",
    modelApiIdentifier,
    modelSelectionField: pluralModelApiIdentifier,
    namespace: null,
    variables: { ids: { required: true, type: "[GadgetID!]" } },
    paramOnlyVariables: [],
    defaultSelection: null
  },
  {
    type: "action",
    operationName: "upsertAnalyticsFIVE",
    operationReturnType: "UpsertAnalyticsFIVE",
    functionName: "upsert",
    namespace: null,
    modelApiIdentifier,
    operatesWithRecordIdentity: false,
    modelSelectionField: modelApiIdentifier,
    isBulk: false,
    isDeleter: false,
    variables: {
      on: { required: false, type: "[String!]" },
      analyticsFIVE: { required: false, type: "UpsertAnalyticsFIVEInput" }
    },
    hasAmbiguousIdentifier: false,
    paramOnlyVariables: ["on"],
    hasReturnType: {
      "... on CreateAnalyticsFIVEResult": { hasReturnType: false },
      "... on UpdateAnalyticsFIVEResult": { hasReturnType: false }
    },
    acceptsModelInput: true,
    hasCreateOrUpdateEffect: true,
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "action",
    operationName: "bulkUpsertAnalyticsFIVEs",
    functionName: "bulkUpsert",
    isBulk: true,
    isDeleter: false,
    hasReturnType: false,
    acceptsModelInput: true,
    operatesWithRecordIdentity: false,
    singleActionFunctionName: "upsert",
    modelApiIdentifier,
    modelSelectionField: pluralModelApiIdentifier,
    namespace: null,
    variables: {
      inputs: { required: true, type: "[BulkUpsertAnalyticsFIVEsInput!]" }
    },
    paramOnlyVariables: ["on"],
    defaultSelection: DefaultAnalyticsFIVESelection
  },
  {
    type: "computedView",
    operationName: "view",
    functionName: "view",
    gqlFieldName: "analyticsFIVEGellyView",
    namespace: null,
    variables: {
      query: { type: "String", required: true },
      args: { type: "JSONObject" }
    }
  }
];
const AnalyticsFIVEManager = buildModelManager(
  modelApiIdentifier,
  pluralModelApiIdentifier,
  DefaultAnalyticsFIVESelection,
  operations
);
export {
  AnalyticsFIVEManager,
  DefaultAnalyticsFIVESelection
};
